activity.go

```diff
@@ -92,11 +122,53 @@ func (w *activityCtxWrapper) Name() string {
 }
 
 func (w *activityCtxWrapper) GetInput(name string) interface{} {
-	return w.ctx.GetInput(name)
+
+	val := w.ctx.GetInput(name)
+
+	// if the input value is complex, we need to modify it
+	if oldMdInput := w.legacyAct.Metadata().Input; oldMdInput != nil {
+		if attr, ok := oldMdInput[name]; ok {
+			if attr.Type() == legacyData.TypeComplexObject {
+				md := ""
+
+				//see if there is a corresponding input schema to construct complex object
+				if sIO, ok := w.ctx.(schema.HasSchemaIO); ok {
+					s := sIO.GetInputSchema(name)
+					if s != nil {
+						md = s.Value()
+					}
+				}
+
+				return &legacyData.ComplexObject{Metadata: md, Value: val}
+			}
+		}
+	}
+
+	return val
 }
 
 func (w *activityCtxWrapper) GetOutput(name string) interface{} {
-	return nil //used for schema
+
+	// if the input value is complex, we need to modify it
+	if oldMdOutput := w.legacyAct.Metadata().Output; oldMdOutput != nil {
+		if attr, ok := oldMdOutput[name]; ok {
+			if attr.Type() == legacyData.TypeComplexObject {
+				md := ""
+
+				//see if there is a corresponding input schema to construct complex object
+				if sIO, ok := w.ctx.(schema.HasSchemaIO); ok {
+					s := sIO.GetOutputSchema(name)
+					if s != nil {
+						md = s.Value()
+					}
+				}
+
+				return &legacyData.ComplexObject{Metadata: md, Value: nil}
+			}
+		}
+	}
+
+	return nil
 }
 
 func (w *activityCtxWrapper) SetOutput(name string, value interface{}) {
```

Comment:
@fm-tibco Legacy activity also set ComplexObject struct to output, so we also need to handle complex object type for SetOutput.